### PR TITLE
Add Script alias

### PR DESCRIPTION
**Pull request by marklundin**
Having some discussions with @willeastcott, with the new ESM Script, we want to keep the default script boilerplate as clean as possible. This PR exports an alias for `ScriptType` named `Script`

```javascript
import { Script } from 'playcanvas';

class Rotator extends Script {
  initialize() {
    ...
  }
}
```

UPDATE:
ScriptType has been added to the deprecation.js file
Improved dicunmentation for Script to include ESM example

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by kpal81xd** [comment]
What was the reason for this?

**Review by mvaligursky — DISMISSED** [review]
Looks good.
Should we rename `ScriptType` to `Script` in that module, and export it as `ScriptType` for compatibility?

**Comment by willeastcott** [comment]
> Should we rename ScriptType to Script in that module, and export it as ScriptType for compatibility?

Yeah, because otherwise, the API ref will continue to show `ScriptType`. The aliasing should be done in the backwards compatibility module.

**Comment by marklundin** [comment]
@willeastcott are we happy to simply rename the class and keep the `script-type.js` filename?

**Comment by mvaligursky** [comment]
Rename filename as well, we keep those in sync.

**Comment by willeastcott** [comment]
Maybe do:

* `script.js` -> `script-create.js` (or `script-register.js`)
* `script-type.js` -> `script.js`

**Comment by willeastcott** [comment]
BTW, we should consider deprecating and hiding `registerScript`.

**Comment by mvaligursky** [comment]
We also need to consider the impact in the Editor. Will the Editor be able to still use at least few previous engine versions? Do we need to add some type checking to it or similar? Maybe it all just works, but we need to check.

**Comment by marklundin** [comment]
> * `script.js` -> `script-create.js` (or `script-register.js`)
> * `script-type.js` -> `script.js`

Makes sense, although strictly speaking this would be a breaking change right?

> We also need to consider the impact in the Editor. Will the Editor be able to still use at least few previous engine versions? Do we need to add some type checking to it or similar? Maybe it all just works, but we need to check.

Yeah it should be fine, editor only uses ScriptType when parsing and `pc.Script === pc.ScriptType`

**Comment by marklundin** [comment]
Actually maybe not a breaking change since we haven't yet [published the ESM build ](https://www.npmjs.com/package/playcanvas?activeTab=code). Mainly thinking about `import { Script } from 'playcanvas/framework/script/script-type.js'

**Comment by marklundin** [comment]
This has not been updated

- Renamed script.js -> script-create.js
- Renamed script-type.js -> script.js
- Updated jsdoc references of {ScriptType} => {Script}

Verified with type docs...
image

*Note that now ScriptType does not exist, it does not generate a corresponding url. It would be great to retain the original, to indicated to users that it has been renamed

File: src/framework/script/script.js [hunk 1]
```diff
@@ -1,156 +1,393 @@
...
+     * @description Called when a Script that already exists in the registry
+     * gets redefined. If the new Script has a `swap` method in its prototype,
+     * then it will be executed to perform hot-reload at runtime.
+     * @param {Script} old - Old instance of the script to copy data to the new instance.
+     */
 }
-/* eslint-enable jsdoc/check-examples */
 
-export { createScript, registerScript, getReservedScriptNames };
+export { Script, Script as ScriptType };
```
Comment (willeastcott) [inline src/framework/script/script.js:393]: I would handle the aliasing in `deprecated.js`.

Comment (marklundin) [reply src/framework/script/script.js:393]: You mean adding `export { Script as ScriptType }` to deprecated.js?

Comment (willeastcott) [reply src/framework/script/script.js:393]: I was thinking something like: https://github.com/playcanvas/engine/blob/c5177982f9c9064df1d5a881a513a302054d60ae/src/deprecated/deprecated.js#L725

File: src/framework/script/script.js [hunk 2]
```diff
@@ -7,11 +7,20 @@ import { ScriptAttributes } from './script-attributes.js';
...
+ * from the engine.
  *
- * The type is to be extended using its JavaScript prototype. There is a list of methods that will
- * be executed by the engine on instances of this type, such as:
+ * You can create a Script using either {@link createScript} or by extending the class directly.
+ *
+ * ```javascript
+ * class Rotator extends Script {
+ *    update() {
+ *      this.entity.rotate(0, 0.1, 0);
```
Comment (Maksims) [inline src/framework/script/script.js:20]: Should be 4 spaces instead of 2.

File: src/framework/script/script-create.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
+import { Debug } from '../../core/debug.js';
+import { EventHandler } from '../../core/event-handler.js';
+
+import { script } from '../script.js';
+import { AppBase } from '../app-base.js';
+
+import { ScriptAttributes } from './script-attributes.js';
+import { ScriptType } from './script.js';
```
Comment (willeastcott) [inline src/framework/script/script-create.js:8]: `ScriptType` -> `Script`

File: src/framework/script/script-create.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        Debug.error('This project is using the legacy script system. You cannot call pc.createScript().');
+        return null;
+    }
+
+    if (reservedScriptNames.has(name))
+        throw new Error(`Script name '${name}' is reserved, please rename the script`);
+
+    const scriptType = function (args) {
+        EventHandler.prototype.initEventHandler.call(this);
+        ScriptType.prototype.initScriptType.call(this, args);
```
Comment (willeastcott) [inline src/framework/script/script-create.js:70]: `ScriptType` -> `Script`

**Comment by marklundin** [comment]
jsdocs update

image

**Comment by marklundin** [comment]
Any more updates on this @willeastcott? Keen to get this merged

**Pull request merged**